packages/app/hooks/useMaybeChangeIcon/index.ts

```diff
@@ -0,0 +1,9 @@
+import { useNavigation } from "@react-navigation/native";
+
+export function useMaybeChangeIcon(changeCopiedFunc: (arg1: boolean) => void) {
```

Comment:
I'd make it more generic so we could reuse it in the future. For example, `useNavigationEvent` that will consume `eventName` and `eventHandler`. You should also remember to register listeners inside `useEffect` hook, otherwise, a new listener will be registered on every render. Then you should remove the listener in `useEffect` cleanup function.